### PR TITLE
feat(remix-node): replace extended classes for `NodeRequest` & `NodeResponse` with interface type casts

### LINKED ISSUE #4395: fetch() doesn't return correct Response type
### What version of Remix are you using?

1.7.2

### Steps to Reproduce

Inside the remix project try

```javascript
const a = await fetch('https://google.com')
console.log('a instanceof Response: ', a instanceof Response) // -> false
```

This probably has to do with this commit: 
https://github.com/remix-run/remix/commit/25b4c83b10c5bc4d84ea81407e7de1a0e4f43df4#diff-87ec1c3c130fe3481b9f1c542b1d8a53548882d33c73c68b894fd442d606afc0

A current workaround I found thanks to @milamer is to include 
```javascript
import { Response } from '@remix-run/web-fetch'
```
in the file.

### Expected Behavior

instanceof Response should be true

### Actual Behavior

instanceof Response is false

### LINKED ISSUE #7059: `fetch(...) instanceof Response` fails after installGlobals()
### What version of Remix are you using?

1.19

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Many libraries depend `resp instanceof Response` to return true; https://github.com/panva/oauth4webapi/blob/170f44e2f7c7da60feb8a654871c56fc0a741e6b/src/index.ts#L1513 is one of such a library. However, after Remix patches the global `fetch` instance, this no longer holds.

I've  tested this with the following three scenarios:

1. vanilla environment in node 20 (works)
2. remix-run/web-fetch (works)
3. remix-run/node (doesn't work)

```
{
  // correct behavior
  console.log((await fetch("http://example.com")) instanceof Response);
}
{
  // also correct behavior
  const i = await import("@remix-run/web-fetch");
  console.log((await i.fetch("http://example.com")) instanceof i.Response);
}
{
  // incorrect behavior; fails.
  const j = await import("@remix-run/node");
  console.log((await j.fetch("http://example.com")) instanceof j.Response);
}
```

The last one is important, as this is the version of globals that's installed via `installGlobals`.

### Expected Behavior

`(await fetch(...)) instanceof Response` should return `true` for fetch and Response exposed by `remix-run/node`.

### Actual Behavior

as the exposed types from `remix-run/node` is `NodeResponse`, the `instanceof` fails.

### ❓ Question 

What I don't particularly understand is why we have the `NodeRequest` and `NodeResponse`. This is the reason why it's causing this type mismatch issue; these classes seem to be serving very little purpose. We're actually not instantiating these types at all, most of the (non-test) code is using `... as NodeResponse` to coerce types. Can we just eliminate them, and expose `@remix-run/web-fetch` classes and types directly?

### PR DESCRIPTION
**Pull request by jeeyoungk**
Closes #3480
Closes #4148
Closes #4395
Closes #7059

This is a fix for the referenced issue; The fix is by

* Replace `NodeRequest` and `NodeResponse` classes with interfaces.
* However, class is still needed (as people may want to do `new NodeRequest` or `x instanceof NodeRequest`). For this purpose, `WebRequest` and `WebResponse` are exposed as classes.

This code is quite tricky as we're exporting types and values separately. The correct way of doing this is to define a type and value with the same identifier, `export` handles this correctly.

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

**Comment by jacob-ebey** [comment]
Closes #7059 #4148 #4395 #3480

**Review by jacob-ebey — APPROVED** [review]

File: .changeset/global-fetch-instanceof.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"remix": patch
```
Comment (MichaelDeBoey) [inline .changeset/global-fetch-instanceof.md:2]:
```suggestion
```

File: integration/fetch-globals-test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import type { Fixture, AppFixture } from "./helpers/create-fixture";
+import { createAppFixture, createFixture, js } from "./helpers/create-fixture";
+
+let fixture: Fixture;
+let appFixture: AppFixture;
+
+test.beforeAll(async () => {
+  fixture = await createFixture({
+    files: {
+      "app/routes/index.jsx": js`
```
Comment (MichaelDeBoey) [inline integration/fetch-globals-test.ts:12]:
```suggestion
      "app/routes/index.tsx": js`
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Comment by jacob-ebey** [comment]
I've re-targeted this PR to the `dev` branch here: https://github.com/remix-run/remix/pull/7109

**Pull request closed without merging**